modules/system/classes/UpdateManager.php

```diff
@@ -533,11 +533,14 @@ public function updatePlugin($name)
     }
 
     /**
-     * Removes an existing plugin
+     * Rollback an existing plugin
+     *
      * @param string $name Plugin name.
+     * @param string $stopOnVersion If this parameter is specified, the process stops after
+     * the specified version is rolled back.
      * @return self
      */
-    public function rollbackPlugin($name)
+    public function rollbackPlugin(string $name, string $stopOnVersion = null): self
```

Comment:
I would prefer we avoid using `: self` return types (and most return types for now to be perfectly honest - not convinced of their value 100%). See https://stackoverflow.com/questions/39068983/php-7-interfaces-return-type-hinting-and-self for more information

Reply:
@LukeTowers Interesting - I have not had any trouble with `self` return types before, but the answers in that SO question do make sense. I feel return types are a net positive, especially with enforcing a certain system structure, but happy to not include them for now if they're going to be problematic.

Reply:
I'm not opposed to basic return types in some cases, but so far my experience with them has been more painful and irritating than helpful (See PHPUnit's :void fiasco). Parameter type hints are welcome though.

Reply:
I removed the precautionary return type